src/renderer/index.jsx

```diff
@@ -18,202 +17,144 @@ if (process.env.NODE_ENV === 'production') {
 // window.eval = global.eval = () => { // eslint-disable-line no-multi-assign, no-eval
 //   import 'bootstrap/dist/css/bootstrap.min.css';
 //     throw new Error(`Sorry, ${remote.app.name} does not support window.eval() for security reasons.`);
-//   // eslint-disable-next-line import/no-unresolved
 //   };
 
 import url from 'url';
 
 import React from 'react';
 import ReactDOM from 'react-dom';
-import {remote, ipcRenderer} from 'electron';
+import {ipcRenderer} from 'electron';
 
-import Config from '../common/config';
+import {GET_CONFIGURATION, UPDATE_TEAMS, QUIT} from 'common/communication';
 
 import EnhancedNotification from './js/notification';
 import MainPage from './components/MainPage.jsx';
-import {createDataURL as createBadgeDataURL} from './js/badge';
 
 Notification = EnhancedNotification; // eslint-disable-line no-global-assign, no-native-reassign
 
-const config = new Config(remote.app.getPath('userData') + '/config.json', remote.getCurrentWindow().registryConfigData);
-
-const teams = config.teams;
+let config;
+let teams;
+
+const reloadConfig = (newConfig) => {
+  config = newConfig;
+  teams = config.teams;
+};
+
+const requestConfig = async (exitOnError) => {
+  // todo: should we block?
+  try {
+    const configRequest = await ipcRenderer.invoke(GET_CONFIGURATION);
+    reloadConfig(configRequest);
+  } catch (err) {
+    if (exitOnError) {
+      ipcRenderer.send(QUIT, `unable to load configuration: ${err}`, err.stack);
+    }
+  }
+};
 
-remote.getCurrentWindow().removeAllListeners('focus');
+const start = async () => {
```

Comment:
Nit: Move this to the bottom of the file where the rendered component would likely be.